creatures/creatures.cpp

```diff
@@ -930,13 +981,459 @@ int Player::save(bool updateTime, LoadType saveType) {
+    if (isMonster()) {
+        aggro=true;
+
+        if (holdPerson && !monType::isHumanoidLike(getType())) {
+            if (output) {
+                *caster << ColorOn << "^yOnly humanoid, goblinoid, monstrous humanoid, and insectoid creatures are affected by the hold-person spell.\nYour spell dissipated.\n" << ColorOff;
+                broadcast(caster->getSock(), caster->getParent(), "^y%M's spell dissipated.^x", caster.get());
+            }
+            resist=true;
+        } 
+
+        if (holdMonster && isUndead()) {
+            if (output) {
+                *caster << ColorOn << "^yThe undead are unaffected by the hold-monster spell.\nYour spell dissipated.\n" << ColorOff;
+                broadcast(caster->getSock(), caster->getParent(), "^y%M's spell dissipated.^x", caster.get());
+            }
+            resist=true;
+        }
+
+        if (holdUndead && !isUndead()) {
+            if (output) {
+                *caster << ColorOn << "^yOnly the undead are affected by the hold-undead spell!\nYour spell dissipated.\n" << ColorOff;
+                broadcast(caster->getSock(), caster->getParent(), "^y%M's spell dissipated.^x", caster.get());
+            }
+            resist=true;
+        }
+
+        if (holdAnimal && !monType::isAnimal(getType())) {
+            if (output) {
+                *caster << ColorOn << "^yOnly animals, dinosaurs, dire animals, avians, reptiles, and fish are affected by the hold-animal spell.\nYour spell dissipated.\n" << ColorOff;
+                broadcast(caster->getSock(), caster->getParent(), "^y%M's spell dissipated.^x", caster.get());
+            }
+            resist=true;
+        }
+
+        if (holdPlant && !monType::isPlant(getType())) {
+            if (output) {
+                *caster << ColorOn << "^yOnly plant creatures are affected by the hold-plant spell.\nYour spell dissipated.\n" << ColorOff;
+                broadcast(caster->getSock(), caster->getParent(), "^y%M's spell dissipated.^x", caster.get());
+            }
+            resist=true;
+        }  
+        if (holdElemental && !monType::isElemental(getType())) {
+            if (output) {
+                *caster << ColorOn << "^yOnly elemental creatures are affected by the hold-elemental spell.\nYour spell dissipated.\n" << ColorOff;
+                broadcast(caster->getSock(), caster->getParent(), "^y%M's spell dissipated.^x", caster.get());
+            }
+            resist=true;
+        }
+        if (holdFey && !monType::isFey(getType())) {
+            if (output) {
+                *caster << ColorOn << "^yOnly fey creatures are affected by the hold-fey spell.\nYour spell dissipated.\n" << ColorOff;
+                broadcast(caster->getSock(), caster->getParent(), "^y%M's spell dissipated.^x", caster.get());
+            }
+            resist=true;
+        }          
+
+        if(flagIsSet(M_DM_FOLLOW) || isEffected("reflect-magic") || isEffected("fire-sheld")) {
```

Comment:
Fire-shield resists all holds/enchantments? Also typo
```suggestion
        if(flagIsSet(M_DM_FOLLOW) || isEffected("reflect-magic") || isEffected("fire-shield")) {
```

Reply:
Yes. For now. It's all still in playtesting. Nothing set. I'll fix the typo.

Reply:
fixed typo